apps/testingaccessibility/src/utils/algolia/sanity-algolia-client.ts

```diff
@@ -1,9 +1,31 @@
 import algoliaSearchClientCreator from 'algoliasearch'
 import sanityAlgoliaIndexer from 'sanity-algolia'
 import isNull from 'lodash/isNull'
+import {z} from 'zod'
+
+const algoliaWriteKeySchema = z
+  .string()
+  .or(z.undefined())
+  .transform((val, ctx) => {
+    if (process.env.NODE_ENV === 'production' && val === undefined) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message:
+          'ALGOLIA_API_WRITE_KEY must be defined in the production environment.',
+      })
+
+      return z.NEVER
+    }
+
+    return val || ''
+  })
```

Comment:
This is one of the places where Zod's type inference really shines.

The environment variables can be either a string or undefined, so I use a schema to parse this algolia API key.

In the transform, I'm able make the return value into a string which narrows the inferred type. I'm also able to do a runtime check of the value when in production and turn that info a zod error. The inferred type still ends up as `string` (which the `algoliaSearchClientCreator` expects).